### PR TITLE
Make `Host.create` work

### PR DESCRIPTION
**Pull request by Ichimonji10**
According to the current unit test results, `Host.create` frequently fails.
Attempt to fix this. Manual tests shows that the following works on both
upstream and downstream systems:

    >>> from nailgun import entities
    >>> entities.Host().create(create_missing=True)

For more information on the bugs that had to be worked around to make this work,
see:

* https://bugzilla.redhat.com/show_bug.cgi?id=1219653
* https://bugzilla.redhat.com/show_bug.cgi?id=1219654

### CODE REVIEW

**Comment by Ichimonji10** [comment]
This should fix several broken Jenkins tests when NailGun 0.13.0 is released and integrated in to Robottelo. If the three open PRs can be merged, I'm aiming to do exactly that tomorrow.

File: nailgun/entities.py [hunk 1]
```diff
@@ -1148,8 +1148,20 @@ def create_payload(self):
...
+
+        """
+        return Domain(
+            self._server_config,
+            id=self.create_json(create_missing)['id'],
+        ).read()
+
     def read(self, entity=None, attrs=None, ignore=()):
-        """Deal with weirdly named data returned frmo the server.
+        """Deal with weirdly named data returned from the server.
```
Comment (elyezer) [inline nailgun/entities.py]: :+1:

**Comment by elyezer** [comment]
ACK

**Comment by Ichimonji10** [comment]
Rebased.

**Comment by omaciel** [comment]
ACK

**Pull request merged**